### PR TITLE
Add vectorization/simd benchmark

### PR DESCRIPTION
**Pull request by he-andy**

### CODE REVIEW

**Comment by sampsyo** [comment]
Looks good overall. Would you mind if I nitpicked the name? `simd` might not stand the test of time, because many benchmarks could potential exercise SIMD execution. Maybe something like `vsmul`, standing for "vector/scalar multiply"?

File: docs/tools/bench.md [hunk 1]
```diff
@@ -54,8 +54,10 @@ The current benchmarks are:
 * `sum-divisors`: Prints the positive integer divisors of the input integer, followed by the sum of the divisors.
 * `sum-sq-diff`: Output the difference between the sum of the squares of the first *n* natural numbers and the square of their sum.
 * `up-arrow`: Computes [Knuth's up arrow][uparrow] notation, with the first argument being the number, the second argument being the number of Knuth's up arrows, and the third argument being the number of repeats.
+* `simd`: Multiplies a constant to each element of a large array. Tests the performance of vectorization optimizations.
```
Comment (sampsyo) [inline docs/tools/bench.md:57]: Looks like this still needs updating.

File: docs/tools/bench.md [hunk 2]
```diff
@@ -54,8 +54,10 @@ The current benchmarks are:
...
 * `reverse`: Compute number with reversed digits (e.g. 123 -> 321).
 
+
```
Comment (sampsyo) [inline docs/tools/bench.md:60]: No need for an extra line here.

File: bril-rs/rs2bril/README.md [hunk 3]
```diff
@@ -10,7 +10,7 @@ View the interface with `cargo doc --open` or install with `make install` using
 - The `println!` macro is special cased to be converted into a `print` call in Bril. It takes a valid `println!` call, ignores the first argument which it assumes to be a format string, and assumes all of the following comma-separated arguments are variables: `println!("this is ignored{}", these, must, be, variables);`
 - There currently isn't a way to pass arguments to the main function that is also valid Rust. You can either declare arguments in the main function such that it is no longer valid Rust but will generate the correct Bril code, or declared them as const values in the first line and edit the Bril IR later: `fn main(a:i64) {}` or `fn main() {let a:i64 = 0;}`
 - Automatic static memory management <https://www.cs.cornell.edu/courses/cs6120/2020fa/blog/asmm/> has not been implemented so arrays must be explicitly dropped where `drop` is specialized to translate to a call to free: `drop([0]);`
-- For loops, `continue`, `break`, and ranges have not been implemented(but could be).
+- For loops, `continue`, `break`, and ranges have not been implemented(but could be).l
```
Comment (sampsyo) [inline bril-rs/rs2bril/README.md:13]: Looks like an errant insertion.

**Review by sampsyo — COMMENTED** [review]
Looks like some cleanup is required in the docs.

**Comment by he-andy** [comment]
> Looks like some cleanup is required in the docs.

Fixed these, thanks for the feedback!

**Comment by sampsyo** [comment]
Looks good; thanks!

**Review by Scutua — APPROVED** [review]

**Pull request merged**